templates/patterns/complete-patterns/cdk-crud-api-with-event-publisher/lambdas/shared/src/lib.rs

```diff
@@ -0,0 +1,185 @@
+mod view_models;
+mod create_order_command;
+mod update_order_command;
+mod get_order_query;
+mod delete_order_command;
+
+pub use view_models::OrderViewModel;
+pub use create_order_command::{CreateOrderCommand, CreateOrderCommandHandler};
+pub use update_order_command::{UpdateOrderCommand, UpdateOrderCommandHandler};
+pub use get_order_query::{GetOrderQuery, GetOrderQueryHandler};
+pub use delete_order_command::{DeleteOrderCommand, DeleteOrderCommandHandler};
+
+use std::env::VarError;
+use std::fmt;
+use aws_config::meta::region::RegionProviderChain;
+use aws_config::Region;
+use aws_sdk_dynamodb::Client;
+use aws_sdk_dynamodb::error::ProvideErrorMetadata;
+use aws_sdk_dynamodb::types::AttributeValue;
+use uuid::Uuid;
+use thiserror::Error;
+
+#[derive(Debug)]
+pub enum DatabaseKeys {
+    PK,
+    SK,
+    Data,
+    Type
+}
+
+impl fmt::Display for DatabaseKeys {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum ApplicationError {
+    #[error("Order not found with orderId '{0}'")]
+    OrderNotFound(String),
+    #[error("Error with database: {0}")]
+    DatabaseError(String),
+    #[error("Table Name not set")]
+    TableNameNotSet(),
+}
+
+impl From<VarError> for ApplicationError {
+    fn from(_: VarError) -> Self {
+        ApplicationError::TableNameNotSet()
+    }
+}
+
+struct Order {
```

Comment:
Why is this not in it's own file?

Reply:
Yep, fair point. I've put the Order and OrderRepository in the same file so that the Order can keep the fields private.